frontend/src/components/Forms/AddIncidentForm.js

```diff
@@ -1,126 +1,181 @@
-    </Row>
-    <div className="right center-mobile">
-      <Popconfirm
-        placement="topRight"
-        title="Are you sure you want to cancel?"
-        onConfirm={props.closeModal}
-        okText="Yes"
-        cancelText="No"
-      >
-        <Button className="full-mobile" type="secondary">
-          Cancel
-        </Button>
-      </Popconfirm>
-      <Button className="full-mobile" type="primary" htmlType="submit" disabled={props.submitting}>
-        {props.title}
-      </Button>
-    </div>
-  </Form>
-);
+class AddIncidentForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      doDetermination: props.initialValues.determination_type_code,
+    };
+  }
+
+  onDoDeterminationChange = (chars, value) => {
+    this.setState({
+      doDetermination: value,
+    });
+  };
+
+  validateDoSubparagraphs = (value) =>
+    this.state.doDetermination === "DO" && value.length === 0
+      ? "This is a required field"
+      : undefined;
+
+  render() {
+    return (
+      <Form layout="vertical" onSubmit={this.props.handleSubmit(this.props.onSubmit)}>
+        <Row gutter={16}>
+          <Col>
+            <Form.Item>
+              <Field
+                id="incident_timestamp"
+                name="incident_timestamp"
+                label="Incident Date and Time*"
+                placeholder="Please select date and time"
+                component={RenderDate}
+                showTime
+                validate={[required, dateNotInFuture]}
+              />
+            </Form.Item>
+            <Form.Item>
+              <Field
+                id="incident_description"
+                name="incident_description"
+                label="Incident Description*"
+                placeholder=""
+                component={RenderAutoSizeField}
+                validate={[required]}
+              />
+            </Form.Item>
+            <Form.Item>
+              <Field
+                id="determination_type_code"
+                name="determination_type_code"
+                label="Inspector's Determination*"
+                component={renderConfig.SELECT}
+                data={this.props.incidentDeterminationOptions}
+                onChange={this.onDoDeterminationChange}
+                validate={[required]}
+              />
+            </Form.Item>
+            {this.state.doDetermination === "DO" ? (
+              <Form.Item>
+                <Field
+                  id="dangerous_occurrence_subparagraph_ids"
+                  name="dangerous_occurrence_subparagraph_ids"
+                  label="Which sub-paragraph(s) of the code apply to this dangerous occurrence?*"
+                  placeholder="Please choose one or more"
+                  component={renderConfig.MULTI_SELECT}
+                  data={this.props.doSubparagraphOptions}
+                  validate={[this.validateDoSubparagraphs]}
+                />
+              </Form.Item>
+            ) : null}
```

Comment:
If this is only being rendered when `this.state.doDetermination === "DO"`, I would guess that the custom validator is not needed. I'd expect that `required` would work just fine?

Reply:
Good point. The only issue is the value from a multi-select is an array, so `required` wont work. `minLength` technically works but its expecting a string so the error message references characters specifically. So a custom validator is still needed but I can remove the "DO" conditional.

Of note there are currently no other multi-selects that require at least one option to be selected.